src/improvements/template/SetDisputeGameImpl.sol

```diff
@@ -0,0 +1,234 @@
+contract SetDisputeGameImpl is L2TaskBase {
+    using stdToml for string;
+
+    /// @notice Struct representing configuration for the task.
+    struct GameImplConfig {
+        uint256 chainId;
+        address fdgImpl;
+        address pdgImpl;
+        address prevFdgImpl;
+        address prevPdgImpl;
+    }
+
+    /// @notice Mapping of chain ID to configuration for the task.
+    mapping(uint256 => GameImplConfig) public cfg;
+
+    /// @notice Returns the string identifier for the safe executing this transaction.
+    function safeAddressString() public pure override returns (string memory) {
+        return "ProxyAdminOwner";
+    }
+
+    /// @notice Returns string identifiers for addresses that are expected to have their storage written to.
+    function _taskStorageWrites() internal pure virtual override returns (string[] memory) {
+        string[] memory storageWrites = new string[](1);
+        storageWrites[0] = "DisputeGameFactoryProxy";
+        return storageWrites;
+    }
+
+    /// @notice Sets up the template with implementation configurations from a TOML file.
+    function _templateSetup(string memory taskConfigFilePath) internal override {
+        super._templateSetup(taskConfigFilePath);
+        string memory toml = vm.readFile(taskConfigFilePath);
+
+        GameImplConfig[] memory configs = abi.decode(toml.parseRaw(".gameImpls.configs"), (GameImplConfig[]));
+        for (uint256 i = 0; i < configs.length; i++) {
+            cfg[configs[i].chainId] = configs[i];
+        }
+    }
+
+    /// @notice Write the calls that you want to execute for the task.
+    function _build() internal override {
+        // Default values for FDG and PDG bonds
+        uint256 fdgBond = 0.08 ether;
+        uint256 pdgBond = 0.08 ether;
+
+        SuperchainAddressRegistry.ChainInfo[] memory chains = superchainAddrRegistry.getChains();
+        for (uint256 i = 0; i < chains.length; i++) {
+            uint256 chainId = chains[i].chainId;
+            GameImplConfig memory c = cfg[chainId];
+
+            address dgf = superchainAddrRegistry.getAddress("DisputeGameFactoryProxy", chainId);
+            IDisputeGameFactory factory = IDisputeGameFactory(dgf);
+
+            // Set FDG (CANNON) implementation if TOML is different from current on-chain
+            address currentFDG = address(factory.gameImpls(CANNON));
+            if (currentFDG != c.fdgImpl) {
+                factory.setImplementation(CANNON, c.fdgImpl);
+            }
+
+            // Set PDG (PERMISSIONED_CANNON) implementation if TOML is different from current on-chain
+            address currentPDG = address(factory.gameImpls(PERMISSIONED_CANNON));
+            if (currentPDG != c.pdgImpl) {
+                factory.setImplementation(PERMISSIONED_CANNON, c.pdgImpl);
+            }
+
+            // Set FDG bond if not already set or needs update
+            if (fdgBond != 0 && factory.initBonds(CANNON) != fdgBond) {
```

Comment:
nit: since fdgBond is always 0.08 ether, might as well simplify to:
```suggestion
            if (c.fdgImpl != 0 && factory.initBonds(CANNON) != fdgBond) {
```

We also only want to set the initial bond for the specific game types that are affected.